.github/workflows/wheels.yml

```diff
@@ -20,8 +20,40 @@ jobs:
         with:
           python-version: '3.10'
       - uses: dtolnay/rust-toolchain@stable
+        with:
+          components: llvm-tools-preview
       - name: Build wheels
         uses: pypa/cibuildwheel@v2.16.2
+        env:
+          CIBW_BEFORE_BUILD: 'bash ./tools/build_pgo.sh'
+          CIBW_BEFORE_BUILD_WINDOWS: 'bash ./tools/build_pgo.sh && cp /tmp/pgo-data/merged.profdata ~/.'
+          CIBW_ENVIRONMENT: 'RUSTUP_TOOLCHAIN="stable" RUSTFLAGS="-Cprofile-use=/tmp/pgo-data/merged.profdata -Cllvm-args=-pgo-warn-missing-function"'
+          CIBW_ENVIRONMENT_LINUX: 'PATH="$PATH:$HOME/.cargo/bin" CARGO_NET_GIT_FETCH_WITH_CLI="true" RUSTUP_TOOLCHAIN="stable" RUSTFLAGS="-Cprofile-use=/tmp/pgo-data/merged.profdata -Cllvm-args=-pgo-warn-missing-function"'
+          CIBW_ENVIRONMENT_WINDOWS: 'RUSTUP_TOOLCHAIN="stable" RUSTFLAGS="-Cprofile-use=c:\\Users\\runneradmin\\merged.profdata -Cllvm-args=-pgo-warn-missing-function"'
```

Comment:
The reason I did these here was because these variables all have meaning globally for non-tier 1 builds (32bit and non-x86 builds). I tried to get the overload syntax to work in `pyproject.toml` but was having trouble getting a match on more than 1 criteria (e.g. os is linux and arch is x86_64). I found it simpler to just use the env variables to override the environment for the tier 1 builds only.

Reply:
Yeah, that's fair enough in that case.  Can we drop a comment both in this file and in the relevant place in `pyproject.toml` mentioning each other, so we remember to keep stuff in sync if they need to change in the future?

Reply:
Done in: https://github.com/Qiskit/qiskit/pull/11502/commits/856dd53059a65a5f9f1f7bf643dc1b3e7a0c455b